### ISSUE TITLE
Toolbar icon disappears after closing and re-opening window.

### ISSUE THREAD

**Issue by nickbeee** [hultmann/multifox#126]
Firefox 25.0, 26.0 (beta). Multifox 2.0.6, 2.0.7.Windows 7, 8.0, 8.1. Firefox sync enabled.

When I install Multifox, icon appears on toolbar as expected. I then create a new profile (2), Multifox icon does not appear on toolbar. 

I then add the icon from the Firefox toolbar layout and it recognises the correct profile. Close this profile window then re-open it from Multifox icon in default profile - icon is now missing in profile (2). Again, I can add it from toolbar menu and it displays (2) confirming this is the correct profile.

I have tried this from a clean install in case of conflict with other add-ons but this but is still there.

I have looked for my Firefox profiles to see if there are any additional ones created by Multifox. I cannot see any.

**Comment by hultmann** [comment]
Hi

I cannot reproduce the bug. What happens if you add other icons (eg copy/print)?

> I have looked for my Firefox profiles to see if there are any additional ones created by Multifox. I cannot see any.

Multifox doesn't create any files.

**Comment by nickbeee** [comment]
After re-checking it looks like this is W8 W8.1 specific (2.0.6 on W7 is fine). This time when I re-started FF, the icon had vanished from the toolbar on default and all other profiles.

I tried with a print Icon. Shut down each profile window, exited FF and re-started. The icon is still there.

**Comment by hultmann** [comment]
I just tested in Windows 8.1 (Firefox 25.0.1) and Multifox 2.0.7 works like expected (restart, new profile, undo close window).

I'm still looking for a way to reproduce it.

**Comment by hultmann** [comment]
Is Sync always enabled in your tests?

**Comment by nickbeee** [comment]
Yes I'm running sync, but I don't sync add-ons. See attached.

**Comment by nickbeee** [comment]
And these add-ons...

**Comment by nickbeee** [comment]
![addons](https://f.cloud.github.com/assets/5652605/1559644/d4a85d90-4fdd-11e3-8beb-4fe1e756d4fe.jpg)
![firefoxsync](https://f.cloud.github.com/assets/5652605/1559645/d4d09dbe-4fdd-11e3-9079-1f803a1ba8fd.jpg)
I'll try the images again as they don't seem to have made it via email.

**Comment by hultmann** [comment]
Still no luck (same extensions and sync config).

Please move the button to tab toolbar and see if something changes.

**Comment by nickbeee** [comment]
![toolbar01](https://f.cloud.github.com/assets/5652605/1559755/5353dd30-4fe4-11e3-9c9a-bef477546d58.jpg)

In that area (as show above)  - it vanishes as detailed above.

If I create a new toolbar and drag the icon onto it - the icon persists across new profiles and after exiting and re-starting Firefox.

If I place it on the tab toolbar (now I know where it is!) it persists across new profiles and after exiting/re-starting.

![tabtoolbar](https://f.cloud.github.com/assets/5652605/1559767/1d366ae6-4fe5-11e3-817d-659b0e033f7d.jpg)

**Comment by hultmann** [comment]
Please try this version, and copy Browser Console messages (ctrl+shift+J) after opening a new window/profile .

http://code.google.com/p/multifox/downloads/detail?name=multifox-2.0.8b1.321.23-49.xpi&can=2&q=#makechanges

**Comment by hultmann** [comment]
- button in default position

**Comment by nickbeee** [comment]
FIRST SESSION
INSTALL ADD-ON

-- multifox 10:50:43.546 -------------------------------------------------------
setWindowProfile NOP => util.networkListeners.active=false

10:50:43.547 
-- multifox 10:50:43.547 -------------------------------------------------------
insertButton0 null

10:50:43.548 
-- multifox 10:50:43.548 -------------------------------------------------------
insertButton4

10:51:01.851 Key event not available on some keyboard layouts: key="m" modifiers="accel,alt" browser.xul
10:51:01.852 Key event not available on some keyboard layouts: key="g" modifiers="alt control" browser.xul
10:51:09.509 
-- multifox 10:51:09.509 -------------------------------------------------------
getProfileList 3 []

10:51:12.406 
-- multifox 10:51:12.406 -------------------------------------------------------
newIdentity 2

10:51:12.406 
-- multifox 10:51:12.406 -------------------------------------------------------
defineIdentity 2

10:51:12.406 
-- multifox 10:51:12.406 -------------------------------------------------------
save 2

10:51:12.406 
-- multifox 10:51:12.406 -------------------------------------------------------
BrowserWindow.register 2

10:51:12.410 
-- multifox 10:51:12.410 -------------------------------------------------------
networkListeners enable

10:51:12.518 
-- multifox 10:51:12.518 -------------------------------------------------------
insertButton0 null

10:51:12.519 
-- multifox 10:51:12.519 -------------------------------------------------------
insertButton4

10:51:12.656 Expected end of value but found 'solid'.  Error in parsing value for 'border-width'.  Declaration dropped. button.css:16

---

SECOND SESSION - Icon persists in default profile window but does not appear when I open a new profile (2).

-- multifox 10:53:47.560 -------------------------------------------------------
removeEventListener SSTabRestored+SSTabRestoring

10:53:50.065 1384772030064  Sync.Tracker.AdblockPlus    WARN    Changed IDs file adblockplus contains non-object value.

10:53:50.065 1384772030065  Sync.Tracker.AdblockPlus.private    WARN    Changed IDs file adblockplus.private contains non-object value.

10:53:50.168 
-- multifox 10:53:50.168 -------------------------------------------------------
getProfileList 3 []

10:53:51.584 1384772031584  Services.HealthReport.HealthReporter    WARN    No prefs data found.

10:53:52.557 
-- multifox 10:53:52.557 -------------------------------------------------------
newIdentity 2

10:53:52.557 
-- multifox 10:53:52.557 -------------------------------------------------------
defineIdentity 2

10:53:52.557 
-- multifox 10:53:52.557 -------------------------------------------------------
save 2

10:53:52.557 
-- multifox 10:53:52.557 -------------------------------------------------------
BrowserWindow.register 2

10:53:52.561 
-- multifox 10:53:52.561 -------------------------------------------------------
networkListeners enable

10:53:52.669 
-- multifox 10:53:52.669 -------------------------------------------------------
insertButton0 null

10:53:52.669 
-- multifox 10:53:52.669 -------------------------------------------------------
insertButton3

10:53:52.761 Expected end of value but found 'solid'.  Error in parsing value for 'border-width'.  Declaration dropped. button.css:16
10:54:10.233 
-- multifox 10:54:10.233 -------------------------------------------------------
BrowserWindow.unregister 2

10:54:10.233 
-- multifox 10:54:10.233 -------------------------------------------------------
networkListeners disable

10:54:13.430 
-- multifox 10:54:13.430 -------------------------------------------------------
getProfileList 3 []

10:54:16.868 
-- multifox 10:54:16.868 -------------------------------------------------------
newIdentity 2

10:54:16.868 
-- multifox 10:54:16.868 -------------------------------------------------------
defineIdentity 2

10:54:16.869 
-- multifox 10:54:16.869 -------------------------------------------------------
save 2

10:54:16.869 
-- multifox 10:54:16.869 -------------------------------------------------------
BrowserWindow.register 2

10:54:16.873 
-- multifox 10:54:16.873 -------------------------------------------------------
networkListeners enable

10:54:16.975 
-- multifox 10:54:16.975 -------------------------------------------------------
insertButton0 null

10:54:16.975 
-- multifox 10:54:16.975 -------------------------------------------------------
insertButton3

10:54:20.291 
-- multifox 10:54:20.291 -------------------------------------------------------
BrowserWindow.unregister 2

10:54:20.291 
-- multifox 10:54:20.291 -------------------------------------------------------
networkListeners disable

10:54:21.905 Key event not available on some keyboard layouts: key="m" modifiers="accel,alt" browser.xul
10:54:21.905 Key event not available on some keyboard layouts: key="g" modifiers="alt control" browser.xul
10:54:33.915 
-- multifox 10:54:33.915 -------------------------------------------------------
getProfileList 3 []

10:54:35.858 
-- multifox 10:54:35.858 -------------------------------------------------------
newIdentity 2

10:54:35.858 
-- multifox 10:54:35.858 -------------------------------------------------------
defineIdentity 2

10:54:35.858 
-- multifox 10:54:35.858 -------------------------------------------------------
save 2

10:54:35.858 
-- multifox 10:54:35.858 -------------------------------------------------------
BrowserWindow.register 2

10:54:35.862 
-- multifox 10:54:35.862 -------------------------------------------------------
networkListeners enable

10:54:35.962 
-- multifox 10:54:35.962 -------------------------------------------------------
insertButton0 null

10:54:35.962 
-- multifox 10:54:35.962 -------------------------------------------------------
insertButton3

---

THIRD SESSION - no icons visible when profiles restored. Created another profile with Ctrl-Alt-M

-- multifox 10:59:10.154 -------------------------------------------------------
newIdentity 4

10:59:10.154 
-- multifox 10:59:10.154 -------------------------------------------------------
defineIdentity 4

10:59:10.154 
-- multifox 10:59:10.154 -------------------------------------------------------
save 4

10:59:10.154 
-- multifox 10:59:10.154 -------------------------------------------------------
BrowserWindow.register 4

10:59:10.367 
-- multifox 10:59:10.367 -------------------------------------------------------
insertButton0 null

10:59:10.367 
-- multifox 10:59:10.367 -------------------------------------------------------
insertButton3

10:59:10.437 Expected end of value but found 'solid'.  Error in parsing value for 'border-width'.  Declaration dropped. button.css:16

---

**Comment by hultmann** [comment]
Thanks, for some reason 'document.persist(toolbar.id, "currentset")' is being ignored. It should save the button position.

Please try the following:
- create a new profile
- install Multifox (button should appear)
- close Firefox
- open localstorage.rdf (which you can find in the new profile directory).
- please paste it here. Something like

``` xml
<?xml version="1.0"?>
<RDF:RDF xmlns:NC="http://home.netscape.com/NC-rdf#"
         xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-box"
                   sidebarcommand=""
                   width=""
                   src="" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#nav-bar"
                   currentset="unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul">
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#main-window"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#nav-bar"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-box"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-title"/>
  </RDF:Description>
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#main-window"
                   width="1440"
                   height="789"
                   sizemode="normal"
                   screenX="186"
                   screenY="106" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-title"
                   value="" />
</RDF:RDF>
```

**Comment by nickbeee** [comment]
On my system the file is localstore.rdf. Here's the result.

``` xml
<?xml version="1.0"?>
<RDF:RDF xmlns:NC="http://home.netscape.com/NC-rdf#"
         xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <RDF:Description RDF:about="chrome://browser/content/browser.xul">
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#main-window"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#nav-bar"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-box"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-title"/>
  </RDF:Description>
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-title"
                   value="" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#main-window"
                   sizemode="normal"
                   width="1147"
                   height="718"
                   screenX="221"
                   screenY="22" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#nav-bar"
                   currentset="unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-box"
                   sidebarcommand=""
                   width=""
                   src="" />
</RDF:RDF>
```

**Comment by hultmann** [comment]
This just keeps getting weirder and weirder. 

Please try the following:
- create a new Firefox profile
- install Multifox from http://code.google.com/p/multifox/downloads/detail?name=multifox-2.0.8b2.323.23-56.xpi
- close Firefox
- paste localstore.rdf here
- open Firefox (is there a button?)
- create a new window (or identity profile) (is there a button?)
- paste Browser Console messages
- close Firefox
- paste localstore.rdf again

thanks, hopefully it will be the last one.

**Comment by nickbeee** [comment]
New profile, installed Multifox

``` xml
<?xml version="1.0"?>
<RDF:RDF xmlns:NC="http://home.netscape.com/NC-rdf#"
         xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <RDF:Description RDF:about="chrome://mozapps/content/downloads/unknownContentType.xul#unknownContentType"
                   screenX="382"
                   screenY="172" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#nav-bar"
                   currentset="unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-title"
                   value="" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul">
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#main-window"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#nav-bar"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-box"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-title"/>
  </RDF:Description>
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-box"
                   sidebarcommand=""
                   width=""
                   src="" />
  <RDF:Description RDF:about="chrome://mozapps/content/downloads/unknownContentType.xul">
    <NC:persist RDF:resource="chrome://mozapps/content/downloads/unknownContentType.xul#unknownContentType"/>
  </RDF:Description>
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#main-window"
                   width="1202"
                   height="694"
                   sizemode="normal"
                   screenX="2"
                   screenY="31" />
</RDF:RDF>


```

Create new window - browser console messages

```
19:39:03.928 Could not read chrome manifest 'file:///C:/Program%20Files%20(x86)/Mozilla%20Firefox/chrome.manifest'.
19:39:04.012 Could not read chrome manifest 'file:///C:/Program%20Files%20(x86)/Mozilla%20Firefox/browser/extensions/%7B972ce4c6-7e08-4474-a285-3208198ce6fd%7D/chrome.manifest'.
19:39:04.013 While creating services from category 'profile-after-change', could not create service for entry 'Disk Space Watcher Service', contract ID '@mozilla.org/toolkit/disk-space-watcher;1'
19:39:04.500 
-- multifox 19:39:04.500 -------------------------------------------------------
setWindowProfile NOP => util.networkListeners.active=false

19:39:04.792 
-- multifox 19:39:04.792 -------------------------------------------------------
_getToolbarButtons addon-bar <empty>

19:39:04.793 
-- multifox 19:39:04.793 -------------------------------------------------------
_getToolbarButtons developer-toolbar <empty>

19:39:04.793 
-- multifox 19:39:04.793 -------------------------------------------------------
_getToolbarButtons TabsToolbar <empty>

19:39:04.793 
-- multifox 19:39:04.793 -------------------------------------------------------
_getToolbarButtons PersonalToolbar <empty>

19:39:04.793 
-- multifox 19:39:04.793 -------------------------------------------------------
_getToolbarButtons nav-bar unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button

19:39:04.793 
-- multifox 19:39:04.793 -------------------------------------------------------
insertButton0 [[object XULElement] toolbar]

19:39:04.793 
-- multifox 19:39:04.793 -------------------------------------------------------
_getToolbarButtons nav-bar unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button

19:39:04.793 
-- multifox 19:39:04.793 -------------------------------------------------------
insertButton2 nav-bar / null

19:39:09.425 
-- multifox 19:39:09.425 -------------------------------------------------------
getProfileList 1 []

19:39:11.839 
-- multifox 19:39:11.839 -------------------------------------------------------
newIdentity 2

19:39:11.839 
-- multifox 19:39:11.839 -------------------------------------------------------
defineIdentity 2

19:39:11.839 
-- multifox 19:39:11.839 -------------------------------------------------------
save 2

19:39:11.839 
-- multifox 19:39:11.839 -------------------------------------------------------
BrowserWindow.register 2

19:39:11.841 
-- multifox 19:39:11.841 -------------------------------------------------------
networkListeners enable

19:39:12.028 
-- multifox 19:39:12.028 -------------------------------------------------------
_getToolbarButtons addon-bar <empty>

19:39:12.028 
-- multifox 19:39:12.028 -------------------------------------------------------
_getToolbarButtons developer-toolbar <empty>

19:39:12.028 
-- multifox 19:39:12.028 -------------------------------------------------------
_getToolbarButtons TabsToolbar <empty>

19:39:12.029 
-- multifox 19:39:12.028 -------------------------------------------------------
_getToolbarButtons PersonalToolbar <empty>

19:39:12.029 
-- multifox 19:39:12.029 -------------------------------------------------------
_getToolbarButtons nav-bar unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button

19:39:12.029 
-- multifox 19:39:12.029 -------------------------------------------------------
insertButton0 [[object XULElement] toolbar]

19:39:12.029 
-- multifox 19:39:12.029 -------------------------------------------------------
_getToolbarButtons nav-bar unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button

19:39:12.029 
-- multifox 19:39:12.029 -------------------------------------------------------
insertButton2 nav-bar / null

19:39:14.809 1384976354809  Services.HealthReport.HealthReporter    WARN    Saved state file does not exist.

19:39:14.809 1384976354809  Services.HealthReport.HealthReporter    WARN    No prefs data found.

19:39:30.631 Key event not available on some keyboard layouts: key="m" modifiers="accel,alt" browser.xul

```

Close Firefox with Alt-F4

``` xml
<?xml version="1.0"?>
<RDF:RDF xmlns:NC="http://home.netscape.com/NC-rdf#"
         xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-box"
                   sidebarcommand=""
                   width=""
                   src="" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul">
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#main-window"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#nav-bar"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-box"/>
    <NC:persist RDF:resource="chrome://browser/content/browser.xul#sidebar-title"/>
  </RDF:Description>
  <RDF:Description RDF:about="chrome://mozapps/content/downloads/unknownContentType.xul#unknownContentType"
                   screenX="382"
                   screenY="172" />
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#nav-bar"
                   currentset="unified-back-forward-button,urlbar-container,reload-button,stop-button,search-container,webrtc-status-button,bookmarks-menu-button,downloads-button,home-button,window-controls,social-share-button,multifox-button" />
  <RDF:Description RDF:about="chrome://browser/content/devtools/webconsole.xul#devtools-webconsole"
                   screenX="318"
                   screenY="238"
                   width="900"
                   height="350"
                   sizemode="normal" />
  <RDF:Description RDF:about="chrome://browser/content/devtools/webconsole.xul">
    <NC:persist RDF:resource="chrome://browser/content/devtools/webconsole.xul#devtools-webconsole"/>
  </RDF:Description>
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#sidebar-title"
                   value="" />
  <RDF:Description RDF:about="chrome://mozapps/content/downloads/unknownContentType.xul">
    <NC:persist RDF:resource="chrome://mozapps/content/downloads/unknownContentType.xul#unknownContentType"/>
  </RDF:Description>
  <RDF:Description RDF:about="chrome://browser/content/browser.xul#main-window"
                   width="1202"
                   height="694"
                   sizemode="normal"
                   screenX="24"
                   screenY="53" />
</RDF:RDF>
```

**Comment by hultmann** [comment]
Did the button appear after you restart and open a new window?

I'm confused because these results are different from the previous one.

**Comment by nickbeee** [comment]
Yes, the icon appeared after re-opening the second window which looks like
it is behaving as it should.

I'll re-run the tests at the weekend. I am wondering if there is a conflict
with any of my other add-ons, I am thinking of creating a new profile and
adding one-by-one until I reach the same state as my default profile. Maybe
I will be able to break it at some point.

As for now, it is working perfectly with the icon on the tab toolbar. I can
live with that but it would be nice to find out what is causing the
original problem!

**Comment by hultmann** [comment]
Fixed with Firefox 29 Beta + Multifox 2.1.0 (thanks to a new method for adding buttons in Firefox 29+).
